panic: aws_acm_certificate

### Terraform Version
v0.7.9

### Affected Resource(s)
- aws_acm_certificate

### Terraform Configuration Files
```hcl
provider "aws" {
  region = "eu-central-1"
}

// the example used here: https://www.terraform.io/docs/providers/aws/d/acm_certificate.html
data "aws_acm_certificate" "example" {
  domain = "tf.example.com"
  statuses = ["ISSUED"]
}
```

### Panic Output
https://gist.github.com/ligustah/fd63a16e80052ec3fff1a43a9377f6f7

### Expected Behavior
It should not panic

### Actual Behavior
It panicked

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

### Important Factoids
It doesn't appear to require valid AWS credentials, as it crashes before contacting the API.

```
provider "aws" {
  region = "us-east-1"
}

data "aws_acm_certificate" "artifactory" {
  domain = "artifactory.example.com"
  statuses = ["ISSUED"]
}

output "ssl_cert_arn" {
  value = "${data.aws_acm_certificate.artifactory.arn}"
}
```

```
terraform version && terraform plan
Terraform v0.7.10

Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but
will not be persisted to local or remote state storage.

data.aws_acm_certificate.artifactory: Refreshing state...
Error refreshing state: 1 error(s) occurred:

* data.aws_acm_certificate.artifactory: unexpected EOF
panic: interface conversion: interface is []interface {}, not []string
2016/11/10 09:16:14 [DEBUG] plugin: terraform:
2016/11/10 09:16:14 [DEBUG] plugin: terraform: goroutine 116 [running]:
2016/11/10 09:16:14 [DEBUG] plugin: terraform: panic(0x2698420, 0x4250a89c0)
2016/11/10 09:16:14 [DEBUG] plugin: terraform: 	/opt/go/src/runtime/panic.go:500 +0x1a1
2016/11/10 09:16:14 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.dataSourceAwsAcmCertificateRead(0x42548f2c0, 0x2362200, 0x4253c4340, 0x2b00000000000028, 0x0)
2016/11/10 09:16:14 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/builtin/providers/aws/data_source_aws_acm_certificate.go:69 +0x6e6
2016/11/10 09:16:14 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).ReadDataApply(0x42514e600, 0x425007460, 0x2362200, 0x4253c4340, 0x42517b3a8, 0x1, 0x0)
2016/11/10 09:16:14 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/resource.go:207 +0xda
2016/11/10 09:16:14 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).ReadDataApply(0x4254919e0, 0x4253a2000, 0x425007460, 0x0, 0x18, 0x18)
2016/11/10 09:16:14 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/provider.go:317 +0x91
2016/11/10 09:16:14 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).ReadDataApply(0x425146060, 0x425152d60, 0x4251530a0, 0x0, 0x0)
2016/11/10 09:16:14 [DEBUG] plugin: terraform: 	/opt/gopath/src/github.com/hashicorp/terraform/plugin/resource_provider.go:537 +0x4e
2016/11/10 09:16:14 [DEBUG] plugin: terraform: reflect.Value.call(0x4253696e0, 0x4254aa0f8, 0x13, 0x2bfa312, 0x4, 0x425193ed0, 0x3, 0x3, 0x4352fc0, 0x425375b80, ...)
2016/11/10 09:16:14 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:434 +0x5c8
2016/11/10 09:16:14 [DEBUG] plugin: terraform: reflect.Value.Call(0x4253696e0, 0x4254aa0f8, 0x13, 0x425193ed0, 0x3, 0x3, 0x0, 0x425211680, 0x425211620)
2016/11/10 09:16:14 [DEBUG] plugin: terraform: 	/opt/go/src/reflect/value.go:302 +0xa4
2016/11/10 09:16:14 [DEBUG] plugin: terraform: net/rpc.(*service).call(0x424ef3e80, 0x424ef3bc0, 0x424efb770, 0x4250b6300, 0x425146100, 0x2364340, 0x425152d60, 0x16, 0x2364380, 0x4251530a0, ...)
2016/11/10 09:16:14 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:383 +0x148
2016/11/10 09:16:14 [DEBUG] plugin: terraform: created by net/rpc.(*Server).ServeCodec
2016/11/10 09:16:14 [DEBUG] plugin: terraform: 	/opt/go/src/net/rpc/server.go:477 +0x421
2016/11/10 09:16:14 [ERROR] root: eval: *terraform.EvalReadDataApply, err: data.aws_acm_certificate.artifactory: unexpected EOF
2016/11/10 09:16:14 [ERROR] root: eval: *terraform.EvalSequence, err: data.aws_acm_certificate.artifactory: unexpected EOF
2016/11/10 09:16:14 [ERROR] root: eval: *terraform.EvalOpFilter, err: data.aws_acm_certificate.artifactory: unexpected EOF
2016/11/10 09:16:14 [ERROR] root: eval: *terraform.EvalSequence, err: data.aws_acm_certificate.artifactory: unexpected EOF
2016/11/10 09:16:14 [TRACE] [walkRefresh] Exiting eval tree: data.aws_acm_certificate.artifactory
2016/11/10 09:16:14 [DEBUG] plugin: /Users/ddcbdevins/bin/terraform: plugin process exited
2016/11/10 09:16:14 [DEBUG] vertex "output.ssl_cert_arn", got dep: "data.aws_acm_certificate.artifactory"
2016/11/10 09:16:14 [DEBUG] vertex "root", got dep: "output.ssl_cert_arn"
2016/11/10 09:16:14 [DEBUG] vertex "provider.aws (close)", got dep: "data.aws_acm_certificate.artifactory"
2016/11/10 09:16:14 [DEBUG] vertex "root", got dep: "provider.aws (close)"
2016/11/10 09:16:14 [INFO] Starting shadow graph walk: walkRefresh
2016/11/10 09:16:14 [DEBUG] Waiting for shadow graph to complete...
2016/11/10 09:16:14 [DEBUG] vertex 'root.provider.aws': walking
2016/11/10 09:16:14 [DEBUG] vertex 'root.provider.aws': evaluating
2016/11/10 09:16:14 [TRACE] [walkRefresh] Entering eval tree: provider.aws
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalSequence
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalInitProvider
2016/11/10 09:16:14 [DEBUG] root: eval: terraform.EvalNoop
2016/11/10 09:16:14 [DEBUG] root: eval: terraform.EvalNoop
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalOpFilter
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalSequence
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalGetProvider
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalBuildProviderConfig
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalSetProviderConfig
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalOpFilter
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalSequence
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalConfigProvider
2016/11/10 09:16:14 [TRACE] [walkRefresh] Exiting eval tree: provider.aws
2016/11/10 09:16:14 [DEBUG] vertex "data.aws_acm_certificate.artifactory", got dep: "provider.aws"
2016/11/10 09:16:14 [DEBUG] vertex "provider.aws (close)", got dep: "provider.aws"
2016/11/10 09:16:14 [DEBUG] vertex 'root.data.aws_acm_certificate.artifactory': walking
2016/11/10 09:16:14 [DEBUG] vertex 'root.data.aws_acm_certificate.artifactory': evaluating
2016/11/10 09:16:14 [TRACE] [walkRefresh] Entering eval tree: data.aws_acm_certificate.artifactory
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalSequence
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/11/10 09:16:14 [DEBUG] root: eval: terraform.EvalNoop
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalCountFixZeroOneBoundary
2016/11/10 09:16:14 [TRACE] [walkRefresh] Exiting eval tree: data.aws_acm_certificate.artifactory
2016/11/10 09:16:14 [DEBUG] vertex 'root.data.aws_acm_certificate.artifactory': expanding/walking dynamic subgraph
2016/11/10 09:16:14 [TRACE] Graph after step *terraform.ResourceCountTransformerOld:

data.aws_acm_certificate.artifactory - *terraform.graphNodeExpandedResource
2016/11/10 09:16:14 [TRACE] Graph after step *terraform.TargetsTransformer:

data.aws_acm_certificate.artifactory - *terraform.graphNodeExpandedResource
2016/11/10 09:16:14 [TRACE] Graph after step *terraform.RootTransformer:

data.aws_acm_certificate.artifactory - *terraform.graphNodeExpandedResource
2016/11/10 09:16:14 [DEBUG] vertex 'root.data.aws_acm_certificate.artifactory': walking
2016/11/10 09:16:14 [DEBUG] vertex 'root.data.aws_acm_certificate.artifactory': evaluating
2016/11/10 09:16:14 [TRACE] [walkRefresh] Entering eval tree: data.aws_acm_certificate.artifactory
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalSequence
2016/11/10 09:16:14 [DEBUG] root: eval: terraform.EvalNoop
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalInstanceInfo
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalOpFilter
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalSequence
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalWriteState
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalIf
2016/11/10 09:16:14 [DEBUG] root: eval: terraform.EvalNoop
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalGetProvider
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalReadDataDiff
2016/11/10 09:16:14 [DEBUG] root: eval: *terraform.EvalReadDataApply
2016/11/10 09:16:14 [ERROR] root: eval: *terraform.EvalReadDataApply, err: data.aws_acm_certificate.artifactory: unexpected EOF
2016/11/10 09:16:14 [ERROR] root: eval: *terraform.EvalSequence, err: data.aws_acm_certificate.artifactory: unexpected EOF
2016/11/10 09:16:14 [ERROR] root: eval: *terraform.EvalOpFilter, err: data.aws_acm_certificate.artifactory: unexpected EOF
2016/11/10 09:16:14 [ERROR] root: eval: *terraform.EvalSequence, err: data.aws_acm_certificate.artifactory: unexpected EOF
2016/11/10 09:16:14 [TRACE] [walkRefresh] Exiting eval tree: data.aws_acm_certificate.artifactory
2016/11/10 09:16:14 [DEBUG] vertex "output.ssl_cert_arn", got dep: "data.aws_acm_certificate.artifactory"
2016/11/10 09:16:14 [DEBUG] vertex "root", got dep: "output.ssl_cert_arn"
2016/11/10 09:16:14 [DEBUG] vertex "provider.aws (close)", got dep: "data.aws_acm_certificate.artifactory"
2016/11/10 09:16:14 [DEBUG] vertex "root", got dep: "provider.aws (close)"
2016/11/10 09:16:14 [ERROR] Shadow graph error: 1 error(s) occurred:

* data.aws_acm_certificate.artifactory: unexpected EOF
2016/11/10 09:16:14 [DEBUG] plugin: waiting for all plugin processes to complete...



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/builtin/providers/aws/data_source_aws_acm_certificate.go
+++ b/builtin/providers/aws/data_source_aws_acm_certificate.go
@@ -2,7 +2,6 @@ package aws
 
 import (
 	"fmt"
-	"strings"
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
@@ -41,11 +40,7 @@ func dataSourceAwsAcmCertificateRead(d *schema.ResourceData, meta interface{}) e
 	statuses, ok := d.GetOk("statuses")
 	if ok {
 		statusStrings := statuses.([]interface{})
-		statusList := make([]*string, len(statusStrings))
-		for i, status := range statusStrings {
-			statusList[i] = aws.String(strings.ToUpper(status.(string)))
-		}
-		params.CertificateStatuses = statusList
+		params.CertificateStatuses = expandStringList(statusStrings)
 	} else {
 		params.CertificateStatuses = []*string{aws.String("ISSUED")}
 	}
@@ -65,12 +60,10 @@ func dataSourceAwsAcmCertificateRead(d *schema.ResourceData, meta interface{}) e
 	}
 
 	if len(arns) == 0 {
-		return fmt.Errorf("No certificate with statuses [%s] for domain %q found in this region.",
-			strings.Join(statuses.([]string), ", "), target)
+		return fmt.Errorf("No certificate for domain %q found in this region.", target)
 	}
 	if len(arns) > 1 {
-		return fmt.Errorf("Multiple certificates with statuses [%s] for domain %s found in this region.",
-			strings.Join(statuses.([]string), ","), target)
+		return fmt.Errorf("Multiple certificates for domain %q found in this region.", target)
 	}
 
 	d.SetId(time.Now().UTC().String())

--- a/builtin/providers/aws/data_source_aws_acm_certificate_test.go
+++ b/builtin/providers/aws/data_source_aws_acm_certificate_test.go
@@ -2,62 +2,45 @@ package aws
 
 import (
 	"fmt"
-	"os"
+	"regexp"
 	"testing"
 
 	"github.com/hashicorp/terraform/helper/resource"
-	"github.com/hashicorp/terraform/terraform"
 )
 
-func TestAccAwsAcmCertificateDataSource_basic(t *testing.T) {
-	region := os.Getenv("AWS_ACM_TEST_REGION")
-	domain := os.Getenv("AWS_ACM_TEST_DOMAIN")
-	certArn := os.Getenv("AWS_ACM_TEST_CERT_ARN")
+func TestAccAwsAcmCertificateDataSource_noMatchReturnsError(t *testing.T) {
+	domain := "hashicorp.com"
 	resource.Test(t, resource.TestCase{
 		PreCheck: func() {
 			testAccPreCheck(t)
-			if region == "" {
-				t.Skip("AWS_ACM_TEST_REGION must be set to a region an ACM certificate pre-created for this test.")
-			}
-			if domain == "" {
-				t.Skip("AWS_ACM_TEST_DOMAIN must be set to a domain with an ACM certificate pre-created for this test.")
-			}
-			if certArn == "" {
-				t.Skip("AWS_ACM_TEST_CERT_ARN must be set to the ARN of an ACM cert pre-created for this test.")
-			}
 		},
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccCheckAwsAcmCertificateDataSourceConfig(region, domain),
-				Check:  testAccCheckAcmArnMatches("data.aws_acm_certificate.test", certArn),
+				Config:      testAccCheckAwsAcmCertificateDataSourceConfig(domain),
+				ExpectError: regexp.MustCompile(`No certificate for domain`),
+			},
+			{
+				Config:      testAccCheckAwsAcmCertificateDataSourceConfigWithStatus(domain),
+				ExpectError: regexp.MustCompile(`No certificate for domain`),
 			},
 		},
 	})
 }
 
-func testAccCheckAcmArnMatches(name, expectArn string) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		ms := s.RootModule()
-		rs, ok := ms.Resources[name]
-		if !ok {
-			return fmt.Errorf("Not found: %s", name)
-		}
-		gotArn := rs.Primary.Attributes["arn"]
-		if gotArn != expectArn {
-			return fmt.Errorf("Expected cert to have arn: %s, got: %s", expectArn, gotArn)
-		}
-		return nil
-	}
-}
-
-func testAccCheckAwsAcmCertificateDataSourceConfig(region, domain string) string {
+func testAccCheckAwsAcmCertificateDataSourceConfig(domain string) string {
 	return fmt.Sprintf(`
-provider "aws" {
-	region = "%s"
-}
 data "aws_acm_certificate" "test" {
 	domain = "%s"
 }
-	`, region, domain)
+`, domain)
+}
+
+func testAccCheckAwsAcmCertificateDataSourceConfigWithStatus(domain string) string {
+	return fmt.Sprintf(`
+data "aws_acm_certificate" "test" {
+	domain = "%s"
+	statuses = ["ISSUED"]
+}
+`, domain)
 }

